File: examples/flutter/lib/src/models/function_list.dart
part of appwrite.models;

/// Functions List
class FunctionList {
    /// Total number of items available on the server.
    final int sum;
    /// List of functions.
    final List<Function> functions;

    FunctionList({
        required this.sum,
        required this.functions,
    });

    factory FunctionList.fromMap(Map<String, dynamic> map) {
        return FunctionList(
            sum: map['sum'],
            functions: List<Function>.from(map['functions'].map((p) => Function.fromMap(p))),
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "sum": sum,
            "functions": functions.map((p) => p.toMap()),
        };
    }
}

File: examples/swift-server/Sources/AppwriteModels/FunctionList.swift

/// Functions List
public class FunctionList {

    /// Total number of items available on the server.
    public let sum: Int

    /// List of functions.
    public let functions: [Function]

    init(
        sum: Int,
        functions: [Function]
    ) {
        self.sum = sum
        self.functions = functions
    }

    public static func from(map: [String: Any]) -> FunctionList {
        return FunctionList(
            sum: map["sum"] as! Int,
            functions: (map["functions"] as! [[String: Any]]).map { Function.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "sum": sum as Any,
            "functions": functions.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                    
}